src/ol/interaction/Snap.js

```diff
@@ -381,7 +381,8 @@ class Snap extends PointerInteraction {
     if (currentMap) {
       keys.forEach(unlistenByKey);
       keys.length = 0;
-      features.forEach(this.forEachFeatureRemove_.bind(this));
+      this.rBush_.clear();
+      features.forEach(this.unlistenFeatureChange_.bind(this));
```

Comment:
You could speed this up even more by getting rid of the new `unlistenFeatureChange_` method and this change:
```suggestion
      Object.values(this.featureChangeListenerKeys_).forEach(unlistenByKey);
      this.featureChangeListenerKeys_ = {};
```

Reply:
Nice! It has been modified in this way